skbio/sequence/__init__.py

```diff
@@ -24,7 +24,7 @@
 
    Sequence
    IUPACSequence
-   NucleotideSequence
+   NucleotideMixin
```

Comment:
Good question -- we probably don't need to export IUPACSequence either. I was thinking it could be useful if devs wanted to subclass it, but it's probably less confusing to end users if we just hide this intermediate ABC. @ebolyen thoughts?